lib/features/digital_guide_view/data/repository/digital_guide_repository.dart

```diff
@@ -14,6 +14,35 @@ Future<DigitalGuideResponse> getDigitalGuideData(Ref ref, int id) async {
   dio.options.headers["Authorization"] =
       "Token ${Env.digitalGuideAuthorizationToken}";
   final response = await dio.get(digitalGuideUrl);
-
-  return DigitalGuideResponse.fromJson(response.data as Map<String, dynamic>);
+  final digitalGuideResponse = DigitalGuideResponse.fromJson(response.data as Map<String, dynamic>);
+  final imageUrl = await getImageUrl(ref, digitalGuideResponse.images[0]);
+  return digitalGuideResponse.copyWith(imageUrl: imageUrl);
 }
+
+@riverpod
+Future<String> getImageUrl(Ref ref, int id) async {
```

Comment:
We should probably do sth more than just throw exceptions her, it's bad practice. I'd recommend some kind of Loggers. In case of error, img should be just null.

Reply:
yeah, try catching to then just throw again doesn't do much difference for us. We should catch the original (wrapped by the riverpod anyway) errors in the presentation layer and present some Error widget there. Probably best to just use `MyErrorWidget` for now, which has build-in logging and crashlytics

Reply:
Errors don't add any value here as it should be handled later on